pkg/cosign/cosign.go

```diff
@@ -116,17 +125,11 @@ func FetchAttestations(imageRef string, key []byte, repository string) ([]map[st
 
 	cosignOpts := &cosign.CheckOpts{
 		//RootCerts:            fulcio.GetRoots(),
-		ClaimVerifier:        cosign.IntotoSubjectClaimVerifier,
-		SigTagSuffixOverride: cosign.AttestationTagSuffix,
-		SigVerifier:          dsse.WrapVerifier(pubKey),
-		VerifyBundle:         false,
-	}
-
-	if err := setSignatureRepo(cosignOpts, ref, repository); err != nil {
-		return nil, errors.Wrap(err, "failed to set signature repository")
+		ClaimVerifier: cosign.IntotoSubjectClaimVerifier,
+		SigVerifier:   dsse.WrapVerifier(pubKey),
 	}
 
-	verified, err := client.Verify(context.Background(), ref, cosignOpts)
+	verified, _, err := client.Verify(context.Background(), ref, cosign.AttestationsAccessor, cosignOpts)
```

Comment:
We need to handle the COSIGN_REPOSITORY here as well, just like you did in VerifySignatures. We can potentially extract and reuse the logic.

Reply:
Done!

Reply:
Looks good! The test is reporting this error:

> pkg/engine/imageVerify.go:208:59: cannot use ([]byte)(key) (type []byte) as type string in argument to "github.com/kyverno/kyverno/pkg/cosign".FetchAttestations